### PR TITLE
Add addon-a11y to monorepo

### PR DESCRIPTION
**Pull request by ndelangen**
Closes: https://github.com/storybooks/storybook-addon-a11y/issues/14

## What I did
Merged in the a11y addon into the monorepo

## How to test
It's used in the cra-kitchen-sink

### CODE REVIEW

File: addons/a11y/.storybook/components/Button/stories.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { checkA11y } from './../../../src';
+
+import Button from './component';
+
+import Faker from 'faker';
+
+const text = Faker.lorem.words();
+
+storiesOf('<Button />', module)
+  .addDecorator(checkA11y)
```
Comment (danielduan) [inline addons/a11y/.storybook/components/Button/stories.js]: I think these a11y stories should be moved over to CRA so we can test them together.

File: addons/a11y/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+This storybook addon can be helpfull to make your UI components more accessibile.
+
+![](docs/screenshot.png)
+
+## Getting started
+
+First, install the addon.
+
+```sh
+$ npm install -D storybook-addon-a11y
```
Comment (danielduan) [inline addons/a11y/README.md]: new package name?

File: addons/a11y/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import '@storybook/addon-a11y/register';
+```
+
+import the `'checkA11y'` decorator to check your stories for violations within your components.
+
+```js
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { checkA11y } from 'storybook-addon-a11y';
```
Comment (danielduan) [inline addons/a11y/README.md]: same here

File: addons/a11y/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "type": "git",
+    "url": "git+https://github.com/storybooks/storybook.git"
+  },
+  "scripts": {
+    "prepare": "node ../../scripts/prepare.js"
+  },
+  "dependencies": {
+    "@storybook/react": "^3.2.14",
+    "axe-core": "^2.0.7",
+    "prop-types": "^15.5.10"
```
Comment (danielduan) [inline addons/a11y/package.json]: peer dep? also this shouldn't require `@storybook/react`

**Review by danielduan — CHANGES_REQUESTED** [review]
Fix up those references to the old package name and move the package Storybook over to our CRA.

I think it's good to take a pass and see if this work with React 16 too. I don't see anything obvious that would break, but more testing is always good.

File: addons/a11y/.storybook/components/Button/stories.js (hunk 1 above)
Comment (ndelangen) [reply addons/a11y/.storybook/components/Button/stories.js]: agree

File: addons/a11y/.gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+node_modules
+coverage
+dist
```
Comment (Hypnosphi) [inline addons/a11y/.gitignore]: Looks like it's all covered by root .gitignore

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

**Comment by ndelangen** [comment]
Added a simple example to cra-kitchen-sink:

screen shot 2017-11-12 at 00 41 22

File: addons/a11y/.storybook/components/Button/stories.js (hunk 1 above)
Comment (ndelangen) [reply addons/a11y/.storybook/components/Button/stories.js]: Added, will leave the old storybook, because might copy over some more stories later.

**Comment by ndelangen** [comment]
Seems to work fine with React 16.1.x!

File: addons/a11y/register.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// NOTE: loading addons using this file is deprecated!
```
Comment (Hypnosphi) [inline addons/a11y/register.js]: Why?

**Review by danielduan — APPROVED** [review]

**Comment by danielduan** [comment]
Code looks good to go, you'll need to rebase and the tests though.

**Comment by ndelangen** [comment]
Should be OK to merge now, @Hypnosphi doe you agree?

**Review by Hypnosphi — APPROVED** [review]

File: examples/cra-kitchen-sink/src/stories/addon-a11y.stories.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import { storiesOf } from '@storybook/react';
+
+import { checkA11y } from '@storybook/addon-a11y';
+import BaseButton from '../components/BaseButton';
+
+const text = 'Testing the a11y addon';
+
+storiesOf('Addon a11y', module)
+  .addDecorator(checkA11y)
+  .add('Default', () => <BaseButton />)
```
Comment (Hypnosphi) [inline examples/cra-kitchen-sink/src/stories/addon-a11y.stories.js]: This produces a propType warning:
```
Warning: Failed prop type: The prop `label` is marked as required in `BaseButton`, but its value is `undefined`
```

Comment (ndelangen) [reply examples/cra-kitchen-sink/src/stories/addon-a11y.stories.js]: Will fix!

**Pull request merged**